cmd/process-agent/config.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	oconfig "github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func getChecks(sysCfg *sysconfig.Config, oCfg *oconfig.OrchestratorConfig, canAccessContainers bool) (checkCfg []checks.Check) {
+	rtChecksEnabled := !ddconfig.Datadog.GetBool("process_config.disable_realtime_checks")
+	if ddconfig.Datadog.GetBool("process_config.process_collection.enabled") {
+		checkCfg = append(checkCfg, checks.Process)
+	} else {
+		if ddconfig.Datadog.GetBool("process_config.container_collection.enabled") && canAccessContainers {
+			checkCfg = append(checkCfg, checks.Container)
+			if rtChecksEnabled {
+				checkCfg = append(checkCfg, checks.RTContainer)
+			}
+		}
+		if ddconfig.Datadog.GetBool("process_config.process_discovery.enabled") {
+			checkCfg = append(checkCfg, checks.ProcessDiscovery)
+		}
+	}
+
+	// activate the pod collection if enabled and we have the cluster name set
+	if oCfg.OrchestrationCollectionEnabled {
+		if oCfg.KubeClusterName != "" {
+			checkCfg = append(checkCfg, checks.Pod)
+		} else {
+			_ = log.Warnf("Failed to auto-detect a Kubernetes cluster name. Pod collection will not start. To fix this, set it manually via the cluster_name config option")
+		}
+	}
+
+	if sysCfg.Enabled {
+		if _, ok := sysCfg.EnabledModules[sysconfig.ProcessModule]; ok {
+			checks.Process.NotifySysprobeProcessModuleEnabled()
```

Comment:
Originally, the ProcessCheck would look at AgentConfig.EnabledChecks to see if Process Module was there to know if it should ask the system probe for privileged process stats. Since we're removing EnabledChecks, and deprecating AgentConfig there isn't really a good way to let the Process Check know that the check is enabled.
I feel this solution is a bit hacky since we modify the global process check, but I've been going back and forth on whether we should do this, or add a parameter to init for all of the checks. Some input would certainly be helpful here.